src/lib.rs

```diff
@@ -74,49 +90,242 @@
 //! }
 //! ```
 //!
-//! Loom will run the closure many times, each time with a different thread
-//! scheduling. The test is guaranteed to fail.
+//! Loom will run the closure provided to `loom::model` many times over, and each time a different
+//! thread scheduling will be used. That is, one execution will have the spawned thread run after
+//! the load from `v2`, and another will have the spawned thread run before the store to `v2`.
+//! Thus, the test is guaranteed to fail.
 //!
 //! # Writing tests
 //!
-//! Test cases using loom must be fully determinstic. All sources of
-//! non-determism must be via loom types. This allows loom to validate the test
-//! case and control the scheduling.
+//! Test cases using loom must be fully determinstic. All sources of non-determism must be via loom
+//! types so that loom can expose different possible values on each execution of the test closure.
+//! Other sources of non-determinism like random number generation or system calls cannot be
+//! modeled by loom.
```

Comment:
Might be worth calling out that one can mock out things like random number generation or syscalls when running a loom test. The Tokio runtime does this to use loom.